File: Golang/Problem2.go
package main


type Problem2 struct {
	list []int
	expect []int
}

func (p Problem2) Calculate() bool{

	length := len(p.list)
	res := make([]int, length)
	for i := 0; i<len(p.list); i++{
		total := 1

		for j := 0; j<len(p.list); j++{

			if j!=i{
				total = total*p.list[j]
			}


		}

		res[i] = total


	}
	//PrintAllNumbers(res)

	return true;
}

/*
func PrintAllNumbers(numbers []int){
	for _,x := range numbers{
		fmt.Print(x)
	}
	fmt.Println("\n")
}
*/

File: Rust/src/Problem2.rs
pub use problem::problem::ProblemTrait;

pub mod problem2{

    pub struct Problem2Struct {
        pub list: Vec<i32>,
        pub exp: Vec<i32>
    }

    impl super::ProblemTrait for Problem2Struct {

        fn calculate(&self) -> bool {
            let mut slize: [i32; 10] = [0; 10];

            for i in 0..self.list.len() {
                let mut total = 1;
                for j in 0..self.list.len(){
                    if j != i {
                        total = total * self.list[j]

                    }
                }

                slize[i] = total;

            }

            return true;
        }


    }





}